src/reactnativeapp/hooks/useTracer.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+import {
+  CompositePropagator,
+  W3CBaggagePropagator,
+  W3CTraceContextPropagator,
+} from "@opentelemetry/core";
+import { WebTracerProvider } from "@opentelemetry/sdk-trace-web";
+import {
+  BatchSpanProcessor,
+  SimpleSpanProcessor,
+  ConsoleSpanExporter,
+} from "@opentelemetry/sdk-trace-base";
+import { registerInstrumentations } from "@opentelemetry/instrumentation";
+import { getWebAutoInstrumentations } from "@opentelemetry/auto-instrumentations-web";
+import { Resource } from "@opentelemetry/resources";
+import {
+  ATTR_DEVICE_ID,
+  ATTR_OS_NAME,
+  ATTR_OS_VERSION,
+  ATTR_SERVICE_NAME,
+  ATTR_SERVICE_VERSION,
+} from "@opentelemetry/semantic-conventions/incubating";
+import { OTLPTraceExporter } from "@opentelemetry/exporter-trace-otlp-http";
+import getLocalhost from "@/utils/Localhost";
+import { useEffect, useState } from "react";
+import {
+  getDeviceId,
+  getSystemVersion,
+  getVersion,
+} from "react-native-device-info";
+import { Platform } from "react-native";
+import {SessionIdProcessor} from "@/utils/SessionIdProcessor";
+
+const Tracer = async () => {
+  // TODO Should add a resource detector for React Native that provides this automatically
+  const resource = new Resource({
+    [ATTR_SERVICE_NAME]: "reactnativeapp",
+    [ATTR_OS_NAME]: Platform.OS,
+    [ATTR_OS_VERSION]: getSystemVersion(),
+    [ATTR_SERVICE_VERSION]: getVersion(),
+    [ATTR_DEVICE_ID]: getDeviceId(),
+  });
+
+  // TODO Not obvious that the WebTracerProvider can be used for React Native, might be useful to have a thin
+  //  ReactNativeTracerProvider on top of it (or BasicTracerProvider) that makes this clear. Could also add some
+  //  protection against browser specific functionality being added to WebTracerProvider that breaks functionality
+  //  for React Native.
+  //  Alternatively could offer a TracerProvider that exposed a JS interface on top of the OTEL Android and Swift SDKS,
+  //  giving developers the option of collecting telemetry at the native mobile layer
+  const provider = new WebTracerProvider({ resource });
```

Comment:
Hmm, is there anything that would not work correctly when a `BasicTracerProvider` is used? 🤔

Reply:
so I seem to get less information using the `BasicTracerProvider`, e.g. for a request to the cart api I see this in Grafana when using it:
![image](https://github.com/user-attachments/assets/c498735a-bcab-4069-bf2c-c961aa5a0929)

but when using `WebTracerProvider` the trace includes the other services:
![image](https://github.com/user-attachments/assets/56d47f2c-27cd-4ad7-b423-f8519f4f39dc)

I think this might be due to the web one setting up a context manager (https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-web/src/WebTracerProvider.ts#L60) whereas the basic one doesn't include any out of the box?

Reply:
Yes that's likely it `ContextManager` and `Progagator` are missing with `BasicTracerProvider` and have to be registered manually. 

Probably fine to keep `WebTracerProvider` then as a first step as it's easier to use. 👍